projects/platform/src/lib/pipes/call.pipe.ts

```diff
@@ -1,22 +1,25 @@
 import { ChangeDetectorRef, EmbeddedViewRef, Pipe, PipeTransform } from '@angular/core';
 
+type Fn = (...args: any[]) => any;
+type Head<T extends Fn> = Parameters<T>[0];
+type Tail<T extends Fn> = T extends (first: any, ...rest: infer R) => any ? R : never;
+type ClassMethod<C, K extends keyof C> = C[K] extends (...args: infer A) => infer R ? (...args: A) => R : never;
+
 @Pipe({ name: 'call' })
-export class CallPipe implements PipeTransform {
-  context: any;
+export class CallPipe<C> implements PipeTransform {
+  private readonly context: C;
 
-  constructor(cd: ChangeDetectorRef) {
-    this.context = (cd as EmbeddedViewRef<any>).context;
+  constructor(private cd: ChangeDetectorRef) {
```

Comment:
```suggestion
  constructor(embeddedViewRef: EmbeddedViewRef<C>) {
    this.context = embeddedViewRef.context;
```

Reply:
this won’t work in view engine, and ivy is disabled for the `platform`